### PR TITLE
Feature: adds a 16.04 build to jenkinsfile

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

**Comment by lsh-0** [comment]
this problem again:

```
[2019-03-08T03:45:53.822Z] [18.212.203.49] out: ----------
[2019-03-08T03:45:53.822Z] [18.212.203.49] out:           ID: /home/elife/.ssh/
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:     Function: cmd.run
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:         Name: ssh-keygen -t rsa -f /home/elife/.ssh/id_rsa -N ""
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:       Result: False
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:      Comment: Command "ssh-keygen -t rsa -f /home/elife/.ssh/id_rsa -N """ run
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:      Started: 03:44:31.130984
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:     Duration: 69.501 ms
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:      Changes:   
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:               ----------
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:               pid:
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:                   12789
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:               retcode:
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:                   1
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:               stderr:
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:                   gethostname: File name too long
[2019-03-08T03:45:53.823Z] [18.212.203.49] out:               stdout:
```

I'm doing some reading. Looks like we have 63bytes to play with (64th byte is reserved for null terminator).

**Comment by giorgiosironi** [comment]
See https://github.com/elifesciences/elife-metrics-formula/pull/3/files#r263715310 which solves the hostname length problem for this case

**Comment by lsh-0** [comment]
this build failure looks like trouble. the error is:

```
[2019-03-20T05:13:11.216Z] Error: Error applying plan:
[2019-03-20T05:13:11.216Z] 
[2019-03-20T05:13:11.216Z] 1 error(s) occurred:
[2019-03-20T05:13:11.216Z] 
[2019-03-20T05:13:11.216Z] * fastly_service_v1.fastly-cdn: 1 error(s) occurred:
[2019-03-20T05:13:11.216Z] 
[2019-03-20T05:13:11.216Z] * fastly_service_v1.fastly-cdn: 400 - Bad Request:
[2019-03-20T05:13:11.216Z] 
[2019-03-20T05:13:11.216Z]     Title:  Bad request
[2019-03-20T05:13:11.216Z]     Detail: invalid dataset "pr-13-fresh-s1604"
```

and I'm not sure if it's the same "dataset" that is referred to in the error, but there is this:

```
[2019-03-20T05:13:08.733Z]   bigquerylogging.2700731677.dataset:            "" => "pr-13-fresh-s1604"
```

**Comment by lsh-0** [comment]
kicking build

**Comment by giorgiosironi** [comment]
This build shouldn't be reading Fastly credentials from Vault. I think the `standalone` build is trying to create a CDN on top of the EC2 instance, but this is unnecessary as we are testing Salt here. See https://github.com/elifesciences/builder/blob/master/projects/elife.yaml#L1618-L1619 for the solution.

**Comment by lsh-0** [comment]
kicking build.

I'm getting unrelated build failures on fresh 16.04 in vagrant, seeing if Alfred can replicate these

**Comment by lsh-0** [comment]
it took a bit of spelunking but there is no 0.10.4 version for 18.04, only 16.04: https://bintray.com/kong/kong-community-edition-deb/ubuntu/0.10.4

0.11.x has no 18.04 version either. 18.04 support starts at 0.14.0, so we're stuck on 16.04 for the interim

**Comment by lsh-0** [comment]
(sigh) that `fastly: false` trick doesn't appear to be work. I may have misunderstood that link

**Comment by giorgiosironi** [comment]
https://github.com/elifesciences/builder/blob/master/projects/elife.yaml#L477-L482 seems correct with `fastly: false`.

**Comment by giorgiosironi** [comment]
https://github.com/elifesciences/builder/blob/master/projects/elife.yaml#L477-L482 seems correct with `fastly: false`. In fact, it does work on `fresh` but not on the `s1604` variant.

**Comment by giorgiosironi** [comment]
Ah, it fails on `/srv/builder/bldr ensure_destroyed:api-gateway--pr-13-fresh-s1604` (leftover from previous builds?). I need to destroy `api-gateway--pr-13-fresh-s1604` manually to allow this build to run again.

**Comment by giorgiosironi** [comment]
Still failing on `s1604`, but without Vault-related errors; getting better.

**Comment by lsh-0** [comment]
thanks for taking a look. I can see some green builds now

**Comment by lsh-0** [comment]
a fresh 16.04 build in vagrant encountered a bunch of errors from trying to do api-calls against kong before it + nginx proxy were up and running. Fixed that (hopefully) and a big red warning about virtual packages.

if this build is green also, then api-gateway will be up next for upgrading.

File: salt/api-gateway/config/lib-systemd-system-kong.service [hunk 1]
```diff
@@ -7,7 +7,7 @@ WantedBy=multi-user.target
 
 [Service]
 StandardError=syslog
-ExecStart=/usr/local/bin/kong start --nginx-conf /etc/kong/nginx.lua
```
Comment (giorgiosironi) [inline salt/api-gateway/config/lib-systemd-system-kong.service]: how come a systemd configuration was already there in 14.04?

Comment (lsh-0) [reply salt/api-gateway/config/lib-systemd-system-kong.service]: https://github.com/elifesciences/api-gateway-formula/pull/5

this is not my first attempt at upgrading our projects to 16.04

**Pull request merged**